editor/scripts/bundle.py

```diff
@@ -317,10 +317,23 @@ def remove_platform_files_from_archive(platform, jar):
     files = zin.namelist()
     files_to_remove = []
 
-    # find files to remove from libexec/* except dmengine and except in platform folder
+    # find files to remove from libexec/*
     libexec_platform = "libexec/" + platform
     for file in files:
-        if file.startswith("libexec") and not file.startswith(libexec_platform) and not file.endswith("/") and not "dmengine" in file:
+        if file.startswith("libexec"):
+            # don't remove any folders
+            if file.endswith("/"):
```

Comment:
Is this robust enough?
I mean, a folder doesn't actually have "/" at the end, it just so happens to be like that here.
Perhaps `os.path.isdir(path)` is useful here?

Reply:
We are iterating the members inside the zip archive here, and libexec/x86_64-darwin/ is a member of the archive. What we get are ZipInfo objects:

https://docs.python.org/2/library/zipfile.html#zipinfo-objects

It might be that the `ZipInfo.external_attr` value can tell us if it is a directory or a file.